+adi/+internal/XUD1aControl.m

```diff
@@ -32,114 +62,136 @@
+                result = obj.getAttributeRAW('voltage0', 'raw', true, obj.XUD1aCtrlDevice);
             end
         end
         
         function set.RxGainMode(obj, value)
-            obj.setAttributeRAW('voltage0', 'raw', num2str(value), true, obj.XUD1aCtrlDevices);
+            obj.setAttributeRAW('voltage0', 'raw', num2str(value), true, obj.XUD1aCtrlDevice);
         end
         
         function result = get.TXRX0(obj)
             result = false;
-            if ~isempty(obj.XUD1aCtrlDevices)
-                result = obj.getAttributeRAW('voltage1', 'raw', true, obj.XUD1aCtrlDevices);
+            if ~isempty(obj.XUD1aCtrlDevice)
+                result = obj.getAttributeRAW('voltage1', 'raw', true, obj.XUD1aCtrlDevice);
             end
         end
         
         function set.TXRX0(obj, value)
-            obj.setAttributeRAW('voltage1', 'raw', num2str(value), true, obj.XUD1aCtrlDevices);
+            obj.setAttributeRAW('voltage1', 'raw', num2str(value), true, obj.XUD1aCtrlDevice);
         end
         
         function result = get.TXRX1(obj)
             result = false;
-            if ~isempty(obj.XUD1aCtrlDevices)
-                result = obj.getAttributeRAW('voltage2', 'raw', true, obj.XUD1aCtrlDevices);
+            if ~isempty(obj.XUD1aCtrlDevice)
+                result = obj.getAttributeRAW('voltage2', 'raw', true, obj.XUD1aCtrlDevice);
             end
         end
         
         function set.TXRX1(obj, value)
-            obj.setAttributeRAW('voltage2', 'raw', num2str(value), true, obj.XUD1aCtrlDevices);
+            obj.setAttributeRAW('voltage2', 'raw', num2str(value), true, obj.XUD1aCtrlDevice);
         end
         
         function result = get.TXRX2(obj)
             result = false;
-            if ~isempty(obj.XUD1aCtrlDevices)
-                result = obj.getAttributeRAW('voltage3', 'raw', true, obj.XUD1aCtrlDevices);
+            if ~isempty(obj.XUD1aCtrlDevice)
+                result = obj.getAttributeRAW('voltage3', 'raw', true, obj.XUD1aCtrlDevice);
             end
         end
         
         function set.TXRX2(obj, value)
-            obj.setAttributeRAW('voltage3', 'raw', num2str(value), true, obj.XUD1aCtrlDevices);
+            obj.setAttributeRAW('voltage3', 'raw', num2str(value), true, obj.XUD1aCtrlDevice);
         end
         
         function result = get.TXRX3(obj)
             result = false;
-            if ~isempty(obj.XUD1aCtrlDevices)
-                result = obj.getAttributeRAW('voltage4', 'raw', true, obj.XUD1aCtrlDevices);
+            if ~isempty(obj.XUD1aCtrlDevice)
+                result = obj.getAttributeRAW('voltage4', 'raw', true, obj.XUD1aCtrlDevice);
             end
         end
         
         function set.TXRX3(obj, value)
-            obj.setAttributeRAW('voltage4', 'raw', num2str(value), true, obj.XUD1aCtrlDevices);
+            obj.setAttributeRAW('voltage4', 'raw', num2str(value), true, obj.XUD1aCtrlDevice);
```

Comment:
This should be protected by a ConnectedToHardware check

Reply:
Added ConnectedToHardware  checks.